README.md

```diff
@@ -22,15 +21,153 @@ various parts of Emmy library, please refer to additional documentation in the *
+
+# Currently supported cryptographic schemes
+
+Currently two anonymous credentials schemes are supported:
+ 
+ * Pseudonym system [4] (see `crypto/zkp/schemes/pseudonymsys`) (supported in &#8484;<sub>p</sub> and EC groups)
+ * Camenisch-Lysyanskaya anonymous credentials [2][15] (see `crypto/zkp/schemes/cl`) - work in progress
+ 
+Pseudonym system [4] was the first anonymous credential scheme and was superseded by CL scheme [2].
+
+## Camenisch-Lysyanskaya anonymous credentials
+
+What are anonymous credentials:
+
+ * User gets a certificate which contains personal data (name, gender, nationality, age ... ) 
+ * The same certificate can be used to connect to different services (even if the databases are joined service providers 
+ cannot map/link the users)
+ * When connecting to a service, user can choose which data to reveal - some services might require only the possession 
+ of a certificate (e.g. certifying that user paid for something), others might require some subset of data contained in certificate
+ 
+While anonymity is obviously a MUST in e-voting, it might gradually become more important in other scenarios as well: 
+
+ * online subscriptions 
+ * wearable healthcare // diabetes - sending data to a diabetes research team 
+ * vehicle communications - cars sending out data about traffic
+ 
+### CL API
+ 
+Let's say University issues to each student a credential where the following attributes are written: name, gender, age,
+student status (graduate/postgraduate). University has its own public and secret key and plays the role of organization
+(when a student connects to the University, a new `Org` from `crypto/zkp/schemes/cl/org.go` is instantiated and 
+responsible for issuing a credential).
+
+First, a student needs to create a master secret key:
+
+```
+masterSecret := pubKey.GenerateUserMasterSecret() // pubKey is University public key
+```
+
+Now let's say variables `name`, `gender`, `age`, `studentStatus` contain big.Int representations for name, gender,
+age, studentStatus (support for different attribute is on TODO list). A student now creates `CredentialManager` which
+will interact with `Org` to obtain a credential (see test code in `client/cl_test.go`):
+
+```
+knownAttrs := []*big.Int{name, gender, age, studentStatus}
+committedAttrs := []*big.Int{}
+hiddenAttrs := []*big.Int{}
+credManager, err := cl.NewCredentialManager(params, pubKey, masterSecret, knownAttrs, committedAttrs,
+    hiddenAttrs)
+```
+
+Note that `committedAttrs` and `hiddentAttrs` are empty - these are needed for attributes for which the organization
+should know only commitments or nothing at all.
+
+Now a student instantiates a communication client:
+
+```
+clClient, err := NewCLClient(testGrpcClientConn)
+```
+
+Student now asks for credential:
+
+```
+cred, err := clClient.IssueCredential(credManager)
+```
+
+Now a student has a credential which can be used 
+
+
+
+ 
+### How does CL work
+ 
+ 
+
+# TODO
```

Comment:
1. I  don't like some of the bullets. If we want to tell the community what are our plans for the future, we need to be more precise, and focus on the what needs to be done in terms of core functionality. For instance, saying "Fix Camenisch-Shoup verifiable encryption" makes me wonder that it isn't working and therefore shouldn't be offered in the first place. Also, stating "Refactor proofs to set all parameters in a constructor" and direct the reader to a comment in the code feels wrong. This is probably more relevant for us as developers, let's use trello for that :wink:. Also, "Performance optimization" of which parts, can you be more specific?
2. This TODO list should be the last thing in the README.
3. Maybe it would be better to call this TODO list _Roadmap_